torch/_tensor_str.py

```diff
@@ -104,9 +104,9 @@ def __init__(self, tensor):
                 return
 
             # Convert to double for easy calculation. HalfTensor overflows with 1e8, and there's no div() on CPU.
-            nonzero_finite_abs = nonzero_finite_vals.abs().double()
-            nonzero_finite_min = nonzero_finite_abs.min().double()
-            nonzero_finite_max = nonzero_finite_abs.max().double()
+            nonzero_finite_abs = nonzero_finite_vals.abs()
```

Comment:
I don't think this is ok.
You can do `dtype == torch.float if t.is_mps() else torch.double` and `.to(dtype=dtype)`.

Reply:
Is the removal of double precision because MPS doesn't support double?

Reply:
Thanks @albanD , updated. That was a lazy local change to get `print()` working.
@ezyang , that's right. Metal itself doesn't have `double` support.